packages/lumx-core/src/scss/components/progress/_index.scss

```diff
@@ -1,21 +1,46 @@
+@use "sass:map";
+
 /* ==========================================================================
    Progress
    ========================================================================== */
 
 .#{$lumx-base-prefix}-progress {
-    &--variant-circular {
-        width: $lumx-progress-circular-size;
-        height: $lumx-progress-circular-size;
+    @each $key, $size in $lumx-sizes {
+        &--variant-circular &--size-#{$key} {
+            width: map.get($lumx-sizes, $key);
+            height: map.get($lumx-sizes, $key);
+        }
     }
 }
 
 /* Circular
    ========================================================================== */
 
 .#{$lumx-base-prefix}-progress-circular {
+    $self: &;
+
     position: relative;
-    width: $lumx-progress-circular-size;
-    height: $lumx-progress-circular-size;
+
+    @each $key, $size in $lumx-sizes {
+        &--size-#{$key} {
+            width: map.get($lumx-sizes, $key);
+            height: map.get($lumx-sizes, $key);
+
+            #{$self}__svg {
+                position: absolute;
+                top: 0;
+                right: 0;
+                bottom: 0;
+                left: 0;
+                display: var(--lumx-material-progress-rotate);
+                width: map.get($lumx-sizes, $key);
```

Comment:
Keep the global definition of svg and only define in the map things related to the size to avoid code duplication